<!DOCTYPE html>
<html lang="en">
<title>scEChIA</title>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1">
<link rel="stylesheet" href="https://www.w3schools.com/w3css/4/w3.css">
<link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Lato">
<link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Montserrat">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
<link rel="stylesheet" href="assets/css/main.css" />
<link rel="stylesheet" href="assets/menustyles.css" />
<script src="http://code.jquery.com/jquery-latest.min.js" type="text/javascript"></script>
<script src="script.js"></script>
<style>
body,h1,h2,h3,h4,h5,h6 {font-family: "Lato", sans-serif}
.w3-bar,h1,button {font-family: "Montserrat", sans-serif}
.fa-anchor,.fa-coffee {font-size:200px}
#main{
  padding-right: 220px;
  padding-left: 220px;}
</st
</style>
<font face="arial" color="black">
<body>

<!-- Navbar -->
<div class="w3-top">
  <div class="w3-bar w3-green w3-card w3-left-align w3-large">
    <a class="w3-bar-item w3-button w3-hide-medium w3-hide-large w3-right w3-padding-large w3-hover-white w3-large w3-green" href="javascript:void(0);" onclick="myFunction()" title="Toggle Navigation Menu"><i class="fa fa-bars"></i></a>
    <a href="index.html" class="w3-bar-item w3-button w3-padding-large w3-white">Home</a>
    <!----<a href="documentation.html" class="w3-bar-item w3-button w3-hide-small w3-padding-large w3-hover-white">Documentation</a>--->
    <!--<a href="faq.html" class="w3-bar-item w3-button w3-hide-small w3-padding-large w3-hover-white">FAQ</a>-->
    <a href="contact.html" class="w3-bar-item w3-button w3-hide-small w3-padding-large w3-hover-white">Contact</a>
  </div>

  <!-- Navbar on small screens -->
  <div id="navDemo" class="w3-bar-block w3-white w3-hide w3-hide-large w3-hide-medium w3-large">
    <!----<a href="documentation.html" class="w3-bar-item w3-button w3-padding-large">Documentation</a>---->
    <!--<a href="faq.html" class="w3-bar-item w3-button w3-padding-large">FAQ</a>-->
    <a href="contact.html" class="w3-bar-item w3-button w3-padding-large">Contact</a>
  </div>
</div>

<!-- Header -->
<header class="w3-container w3-white w3-center" style="padding:50px 8px">
  <img style="width:70%;height:30%;" src="scEChIA.jpg" alt="no">
  <!----<button class="w3-button w3-black w3-padding-large w3-large w3-margin-top">Get Started</button>--->
</header>

<!-- First Grid -->
<section class="wrapper style3" id="main">
	<div class="inner">
		<div class="col col1">
			<h3 style="color:green;">Introduction </h3>
			<p>scEChIA (Single Cell Epigenome Chromatin Interaction Analysis) is a computational method that can predict chromatin interactions among distal sites with high accuracy. scEChIA can also make UCSC track on predicted interactions that may be useful for genomics studies based on chromatin interactions using the UCSC Genome Browser.</p>

<h3 style="color:green;">Installation</h3>
	<h3 class="w3-padding-12" style="color:blue;"><a href="http://192.168.2.225:1207/scEChIA/scEChIA_Package/scEChIA_0.1.0.tar.gz">Download Package</a></h3>
	<p>or use the link "http://192.168.2.225:1207/scEChIA/scEChIA_Package/scEChIA_0.1.0.tar.gz"</p>
	<li>version.string R version 3.6.0</li>
	<li>These libraries need to be installed before the package is installed : varbvs, pracma, glassoFast, nor1mix, roxygen2. </li>
	<li>install.packages("scEChIA_0.1.0.tar.gz", repos=NULL, type="source")</li><br>
<br>

<h3 style="color:green;">Library required for scEChIA package:</h3>
<li>library(glassoFast)</li>
<li>library(nor1mix)</li>
<li>library(roxygen2)</li>
<li>library(scEChIA)</li>
<li>library(varbvs)</li>
<li>library(pracma)</li>
<br>

<br>
<h3 style="color:green;">Chromatin interaction prediction using prior rho information:</h3>
<p>Function_name: Interaction_Prediction_1</p>
<li>Interaction_Prediction_1 can predict interaction with the help of two different cell types' rho value. First we will pass HiC data from two different celltypes and take average of both HiC data to define as final rho matrix for predicting interaction from single cell data.</li>
<br>
<p>How to run:</p>
<pre><code><font face="arial" color="black">setwd(system.file("data", package="scEChIA")) #Keep your input data format like example data<br>genomic_region1 = read.table('K562_HiC_chr17.txt')<br>genomic_region1[is.na(genomic_region1)] = 0<br>genomic_region2 = read.table('IMR90_HiC_chr17.txt')<br>genomic_region2[is.na(genomic_region2)] = 0<br>gap = 25000 #Bin size 25kb (User can change it as per his choice)<br>patternf = 17 #Chromosome number (User can change it according to chromosome number)<br>chrNo = patternf<br>data = read.table('GM12878_SC_chr17.txt')<br>chrinfo = data[, 1:3] #Chromosome location (chr, start, end)<br>chrNo = patternf<br>startCell = 96 #start sample column<br>endCell = 575 #end sample column<br>chromSize = 83257441 #Size of chromosome 17 (Change it according to chromosome number. Follow heading Chromosome Size of hg19)<br>rhomatrix = rhomatAvg(genomic_region1, genomic_region2, gap, patternf, data, chrinfo)<br>predicted_interaction = Interaction_Prediction_1(chrinfo, data, rhomatrix, chrNo, startCell, endCell, chromSize)
</pre></code></font>
<br>
<h3 style="color:green;">Chromatin interaction prediction using constant rho:</h3>
	<p>Function_name: Interaction_Prediction_2</p>
<li>Interaction_Prediction_2 can predict interaction with the help of constant rho value.</li>
<br>
<p>How to run:</p>
<pre><code><font face="arial" color="black">gap = 25000 #Bin size 25kb (User can change it as per his choice)<br>patternf = 17 #Chromosome number (User can change it according to chromosome number)<br>chrNo = patternf<br>data = read.table('GM12878_SC_chr17.txt')<br>chrinfo = data[, 1:3] #Chromosome location (chr, start, end)<br>chrNo = patternf<br>startCell = 96 #start sample<br>endCell = 575 #end sample<br>chromSize = 83257441 #Size of chromosome 17 (Change it according to chromosome number. Follow heading Chromosome Size of hg19)<br>rhoConstant = 0.01 #user can change the value of rhoConstant<br>predicted_interaction = Interaction_Prediction_2(chrinfo, data, rhoConstant, chrNo, startCell, endCell, chromSize)</pre></code></font>
<br>
<h3 style="color:green;">Visualizing chromatin interactions using UCSC track</h3>
<p>Function_name: ucscTrack</p>
<li>Function ucscTrack can create track file for UCSC Genome Browser on predicted interactions. Using addscore value it will provide perfect interaction score in decimal numbers.</li>

<br>
<p>How to Run :</P>
<pre><code><font face="arial" color="black">predInteraction = predicted_interaction<br>addscore = 500 #User can increase or decrease addscore value as per his choice<br>ucscTrack_file = ucscTrack(predInteraction, addscore)<br>
</pre></code></font>
<br>

<h3 style="color:green;">Note : </h3>
 <p class="w3-padding-12">After generating the UCSC track file, the bottom line has to be pasted on the header to be visualized on the UCSC Genome Browser.
track type=interact name="User and celltype" description="User and celltype" interactDirectional=true maxHeightPixels=200:100:50 visibility=full</p>

<p>#chrom chromStart chromEnd name score value exp color sourceChrom sourceStart sourceEnd sourceName sourceStrand targetChrom targetStart targetEnd targetName targetStrand</p><br>

<h4 style="color:green;">Visualization figure of microglia using UCSC Genome Browser (https://genome.ucsc.edu/) : </h4>
<!-- Header -->
<header class="w3-container w3-white w3-center" style="padding:50px 8px">
  <img style="width:75%;height:25%;" src="ucsctrack_manual_page.jpg" alt="no">
  <!----<button class="w3-button w3-black w3-padding-large w3-large w3-margin-top">Get Started</button>--->
</header>

<h4 style="color:green;">How to validate predicted chromatin interactions? : </h4>
<p> 1. Predict the chromatin interaction using scATAC-seq data from scEChIA.
2. Find the related available chromatin interaction from HiC/chromatin-based data and make it paired for every interaction.
3. Use freely available resource - paired genomic loci tool ( pgltools - https://github.com/billgreenwald/pgltools) to intersect predicted chromatin interactions with existing chromatin interactions. 
4. Check the fraction/number of overlap. </p>
<br>

<h3 style="color:green;">Chromosome Size of hg19</h3>
<p class="w3-padding-12">chr1: 248956422, chr2: 242193529, chr3: 198295559, chr4: 190214555, chr5: 181538259, chr6: 170805979, chr7: 159345973, chr8: 145138636, chr9: 138394717, chr10: 133797422, chr11: 135086622, chr12: 133275309, chr13: 114364328, chr14: 107043718, chr15: 101991189, chr16: 90338345, chr17: 83257441, chr18: 80373285, chr19: 58617616, chr20: 64444167, chr21: 46709983, chr22: 50818468, chrX: 156040895, chrY: 57227415</p>
<h3 class="w3-padding-12"><font face="arial" color="blue"><a href="https://github.com/reggenlab/scEChiA/tree/main/scEChIA_Package/Predicted-Chromatin-Interactions">Supplementary Files</a></h3>
</div>
</div>
</section>

<script>
// Used to toggle the menu on small screens when clicking on the menu button
function myFunction() {
  var x = document.getElementById("navDemo");
  if (x.className.indexOf("w3-show") == -1) {
    x.className += " w3-show";
  } else { 
    x.className = x.className.replace(" w3-show", "");
  }
}
</script>

</body>
</html>
